jlexer/lexer.go

```diff
@@ -667,6 +667,8 @@ func (r *Lexer) String() string {
 	}
 	var ret string
 	if r.token.byteValueCloned {
+		r.token.byteValueCloned = false
```

Comment:
Any details on why it fixes the issue? cloned values are immutable and should be safe to return... what is really wrong here?

Actually better to provide details why the issue occurs at all... byteValueCloned should be TRUE only iff the value was allocated and no longer refers to original buffer. So this TRUE path should be safe. Why it turns out it is not?

Reply:
After first optimization *byteValueCloned* stays always *true* and in next property can clone *mutable* value like in #310

Reply:
I think the nicer looking fix is smth like that instead:

```
--- a/jlexer/lexer.go
+++ b/jlexer/lexer.go
@@ -401,6 +401,7 @@ func (r *Lexer) scanToken() {
 // consume resets the current token to allow scanning the next one.
 func (r *Lexer) consume() {
        r.token.kind = tokenUndef
+       r.token.byteValueCloned = false
        r.token.delimValue = 0
 }
```

it guarantees that lexer state is correctly reset in a single place.

Reply:
@kirillx great, did move *r.token.byteValueCloned = false* as your example